File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/Bruker.java
package no.nav.melosys.eessi.controller.dto;

import java.time.LocalDate;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonFormat.Shape;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.Data;

@Data
public class Bruker {

    private String fornavn;
    private String etternavn;

    @JsonDeserialize(using = LocalDateDeserializer.class)
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonFormat(shape = Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate foedseldato;

    private String kjoenn;
    private List<String> statsborgerskap;
    private String fnr;

    private boolean harSensitiveOpplysninger;
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/Bruker.kt
package no.nav.melosys.eessi.controller.dto

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDate

data class Bruker(
    var fornavn: String? = null,
    var etternavn: String? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    var foedseldato: LocalDate, // kaster NullPointerException i Java kode om null
    var kjoenn: String, // kaster NullPointerException i Java kode om null
    var statsborgerskap: Collection<String>, // kaster NullPointerException i Java kode om null
    var fnr: String? = null,
    var harSensitiveOpplysninger: Boolean = false
)
